packages/compass-components/src/components/signal-popover.tsx

```diff
@@ -0,0 +1,389 @@
+          className={signalCardLearnMoreLinkStyles}
+          href={learnMoreLink}
+          target="_blank"
+        >
+          {learnMoreLabel ?? 'Learn more'}
+        </Link>
+      </div>
+    </div>
+  );
+};
+
+const multiSignalHeaderContainerStyles = css({
+  '--multiSignalHeaderBorderColor': palette.gray.light2,
+  '--multiSignalHeaderBackgroundColor': palette.gray.light3,
+  display: 'flex',
+  alignItems: 'center',
+  paddingTop: spacing[1],
+  paddingBottom: spacing[1],
+  paddingLeft: spacing[2],
+  paddingRight: spacing[4],
+  gap: spacing[1],
+  backgroundColor: 'var(--multiSignalHeaderBackgroundColor)',
+  boxShadow: `inset 0 -1px 0 var(--multiSignalHeaderBorderColor)`,
+  fontVariantNumeric: 'tabular-nums',
+});
+
+const MultiSignalHeader: React.FunctionComponent<{
+  currentIndex: number;
+  total: number;
+  onIndexChange(newVal: number): void;
+}> = ({ currentIndex, total, onIndexChange }) => {
+  return (
+    <div className={multiSignalHeaderContainerStyles}>
+      <IconButton
+        data-testid="insight-signal-show-prev-button"
+        aria-label="Show previous insight"
+        title="Show previous insight"
+        onClick={(e) => {
+          e.stopPropagation();
+          onIndexChange(currentIndex - 1);
+        }}
+        disabled={currentIndex === 0}
+      >
+        <Icon glyph="ChevronLeft"></Icon>
+      </IconButton>
+      <span>
+        Insight <strong>{currentIndex + 1}</strong> of <strong>{total}</strong>
+      </span>
+      <IconButton
+        data-testid="insight-signal-show-next-button"
+        aria-label="Show next insight"
+        title="Show next insight"
+        onClick={(e) => {
+          e.stopPropagation();
+          onIndexChange(currentIndex + 1);
+        }}
+        disabled={currentIndex === total - 1}
+      >
+        <Icon glyph="ChevronRight"></Icon>
+      </IconButton>
+    </div>
+  );
+};
+
+const popoverStyles = css({
+  width: 315,
+});
+
+const popoverContentContainerStyles = css({
+  display: 'block',
+});
+
+const transitionStyles = css({
+  transitionProperty: 'opacity, width, border-radius',
+  transitionTimingFunction: 'linear',
+  transitionDuration: '0.15s',
+});
+
+const badgeStyles = css(
+  {
+    // button reset first
+    padding: 0,
+    border: 'none',
+    background: 'none',
+  },
+  {
+    '--badgeBackgroundColor': palette.blue.light3,
+    '--badgeBorderColor': palette.blue.light2,
+    '--badgeColor': palette.blue.dark1,
+    position: 'relative',
+    display: 'inline-block',
+    width: 18,
+    height: 18,
+    color: 'var(--badgeColor)',
+    backgroundColor: 'var(--badgeBackgroundColor)',
+    boxShadow: `inset 0 0 0 1px var(--badgeBorderColor)`,
+    borderRadius: '9px',
+    fontSize: '12px',
+    lineHeight: '14px',
+    fontWeight: 700,
+    overflow: 'hidden',
+    cursor: 'pointer',
+  },
+  transitionStyles
+);
+
+const badgeDarkModeStyles = css({
+  // TODO: https://jira.mongodb.org/browse/COMPASS-6912
```

Comment:
Splitting this out for now as I figured out that I can't find designs in the figma (even though I'm sure there were some?) and I don't want to block this PR. As this is not really in the app and will be under development for a few more days before we start seeing them in the app, I think it's okay